backend/backend/main.py

```diff
@@ -7,15 +7,25 @@
 from core.context import Context
 from utils.logger import logger
 
+from uuid import uuid4
+import os
+
 
 @asynccontextmanager
 async def lifespan(app: FastAPI):
     logger.info("Application starting up...")
+
+    load_dotenv()
```

Comment:
Loading the environment variables twice, once with `load_dotenv()` and then with `load_dotenv("secrets/.env")`, might lead to unexpected behavior if there are overlapping keys. Consider loading them in a specific order or merging them carefully.